share/eds/blockstore.go

```diff
@@ -148,11 +160,30 @@ func (bs *blockstore) getReadOnlyBlockstore(ctx context.Context, cid cid.Cid) (d
 		return nil, fmt.Errorf("failed to find shards containing multihash: %w", err)
 	}
 
+	// a share can exist in multiple EDSes, check cache to contain any of accessors containing shard
```

Comment:
this behavior is incorrect - we only store a single value, since we do not use dagstore's implementation. So we can just use the first element returned. This is related to the badger bloat issue that we discussed in the pruning call.

Reply:
Yes but we don't actually save them in the inverted index - check how our inverted_index.go differs from the default dagstore impl - we only save one shard key per CID, even if we find another shard that the CID points to

Reply:
Ah ok I get it. Is it no really incorrect, it just pointless for loop, that loops over 1 item every time. I've removed for loop and updated comments here.